source/extensions/quic_listeners/quiche/platform/quic_logging_impl.h

```diff
@@ -179,4 +180,19 @@ class QuicLogSink {
 // Return the previous sink.
 QuicLogSink* SetLogSink(QuicLogSink* new_sink);
 
+// Overload std::operator<< to output a vector.
+template <class T> std::ostream& operator<<(std::ostream& out, const std::vector<T>& v) {
+  out << "vector { ";
+  size_t i = 0;
+  for (; i < v.size() - 1; ++i) {
```

Comment:
I think line 186-193 can be replaced by `absl::StrJoin(v, ", ", absl::StreamFormatter())`.